include/seqan3/alignment/pairwise/execution/execution_handler_sequential.hpp

```diff
@@ -0,0 +1,80 @@
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::detail::execution_handler_sequential.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Handles the sequential execution of alignments.
+ * \ingroup execution
+ */
+class execution_handler_sequential
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    execution_handler_sequential()                                                 = default;
+    execution_handler_sequential(execution_handler_sequential const &)             = default;
+    execution_handler_sequential(execution_handler_sequential &&)                  = default;
+    execution_handler_sequential & operator=(execution_handler_sequential const &) = default;
+    execution_handler_sequential & operator=(execution_handler_sequential &&)      = default;
+    ~execution_handler_sequential()                                                = default;
+    //!}
+
+    /*!\name Execution
+     * \{
+     */
+    //!\brief Invokes the passed alignment instance in a blocking manner.
+    template <typename alignment_instance_t,
+              typename delegate_t>
+    void execute(alignment_instance_t && alignment,
+                 delegate_t && delegate)
+    {
+        delegate(std::invoke(alignment));
```

Comment:
why not simply call `alignment()`?

Reply:
That's why:
```
Invoke the Callable object f with the parameters args. As by INVOKE(std::forward<F>(f), std::forward<Args>(args)...).
where INVOKE(f, t1, t2, ..., tN) is defined as follows:
If f is a pointer to member function of class T:
If std::is_base_of<T, std::decay_t<decltype(t1)>>::value is true, then INVOKE(f, t1, t2, ..., tN) is equivalent to (t1.*f)(t2, ..., tN)
If std::decay_t<decltype(t1)> is a specialization of std::reference_wrapper, then INVOKE(f, t1, t2, ..., tN) is equivalent to (t1.get().*f)(t2, ..., tN)
If t1 does not satisfy the previous items, then INVOKE(f, t1, t2, ..., tN) is equivalent to ((*t1).*f)(t2, ..., tN).
Otherwise, if N == 1 and f is a pointer to data member of class T:
If std::is_base_of<T, std::decay_t<decltype(t1)>>::value is true, then INVOKE(f, t1) is equivalent to t1.*f
If std::decay_t<decltype(t1)> is a specialization of std::reference_wrapper, then INVOKE(f, t1) is equivalent to t1.get().*f
If t1 does not satisfy the previous items, then INVOKE(f, t1) is equivalent to (*t1).*f
Otherwise, INVOKE(f, t1, t2, ..., tN) is equivalent to f(t1, t2, ..., tN) (that is, f is a FunctionObject)
```

Reply:
If I don't oversee anything; `std::invoke(callable)` is in all cases equivalent to `callable()`

I would prefer `std::function` in both cases, because it does type erasure and this might have an in pact on compile time (drastically reducing it).

Not important right now.